gp_predict.f95

```diff
@@ -772,9 +772,21 @@ subroutine gpSparse_initialise(this, from, task_manager, condition_number_norm,
 
 #ifdef HAVE_QR
       allocate(c_subYY_sqrtInverseLambda(n_globalSparseX,n_globalY))
+      call matrix_product_vect_asdiagonal_sub(c_subYY_sqrtInverseLambda,from%covariance_subY_y,sqrt(1.0_qp/from%lambda)) ! O(NM)
+      if (allocated(from%covariance_subY_y)) deallocate(from%covariance_subY_y)
```

Comment:
add a comment here to make it explicit we're freeing inside `from`

Reply:
Done.